### PR TITLE
[WFCORE-2589] Eliminating MSC Optional Dependencies in classloading layer

### PR DESCRIPTION
**Pull request by ropalka**

### CODE REVIEW

**Comment by bstansberry** [comment]
What kind of test coverage of optional module dependencies do we have to give us comfort that this isn't going to cause regressions?

**Comment by bstansberry** [comment]
Reverting https://github.com/wildfly/wildfly-core/pull/2871 is one small way of adding coverage.

**Comment by ropalka** [comment]
I'm working on it @bstansberry ...

**Comment by ropalka** [comment]
I reverted https://github.com/wildfly/wildfly-core/pull/2871 @bstansberry and I believe this resuscitated
test case should cover the most significant scenarios because it tries to redeploy multiple times modules with optional dependencies. I also provided fix for WFCORE-2235 issue.

**Comment by ropalka** [comment]
Further more I guarantee we're compliant with previous wildfly releases with these changes applied.

File: testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/deployment/InterdependentDeploymentTestCase.java [hunk 1]
```diff
@@ -295,42 +273,30 @@ private String getJBossDeploymentStructure(String... dependencies) {
...
     public static class ServiceActivatorDeploymentC extends ServiceActivatorDeployment {
-
         public ServiceActivatorDeploymentC() {
             super(ServiceName.of(ServiceActivatorDeployment.class.getSimpleName(), "c"), "c.properties");
         }
     }
 
     public static class ServiceActivatorDeploymentD extends ServiceActivatorDeployment {
-
-        private final ServiceActivatorDeploymentB imported = new ServiceActivatorDeploymentB();
```
Comment (bstansberry) [inline testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/deployment/InterdependentDeploymentTestCase.java]: Why is this removed?

File: testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/deployment/InterdependentDeploymentTestCase.java [hunk 2]
```diff
@@ -295,42 +273,30 @@ private String getJBossDeploymentStructure(String... dependencies) {
...
-
         public ServiceActivatorDeploymentD() {
             super(ServiceName.of(ServiceActivatorDeployment.class.getSimpleName(), "d"), "d.properties");
         }
     }
 
     public static class ServiceActivatorDeploymentE extends ServiceActivatorDeployment {
-
-        private final ServiceActivatorDeploymentD imported = new ServiceActivatorDeploymentD();
```
Comment (bstansberry) [inline testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/deployment/InterdependentDeploymentTestCase.java]: Why is this removed?

File: testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/deployment/InterdependentDeploymentTestCase.java [hunk 3]
```diff
@@ -295,42 +273,30 @@ private String getJBossDeploymentStructure(String... dependencies) {
...
-
         public ServiceActivatorDeploymentE() {
             super(ServiceName.of(ServiceActivatorDeployment.class.getSimpleName(), "e"), "e.properties");
         }
     }
 
     public static class ServiceActivatorDeploymentF extends ServiceActivatorDeployment {
-
-        //private final ServiceActivatorDeploymentC importedC = new ServiceActivatorDeploymentC();
-        private final ServiceActivatorDeploymentD importedD = new ServiceActivatorDeploymentD();
```
Comment (bstansberry) [inline testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/deployment/InterdependentDeploymentTestCase.java]: Why is this removed?

**Comment by ropalka** [comment]
Good catch @bstansberry ! It was oversight on my side. I eliminated them because those were unused fields but I didn't notice those unused fields were imports from other modules. Fixed.

**Comment by ropalka** [comment]
Retest this please

**Comment by ropalka** [comment]
Updated @wildfly/mergers-core

**Comment by jmesnil** [comment]
retest this please
